scripts/manage_demux_stats.py

```diff
@@ -222,17 +271,22 @@ def set_sample_values(demux_process, parser_struct, proc_stats):
                         
             try: 
                 target_file.put()
-            except:
-                problem_handler("exit", "Failed to apply artifact data to LIMS")
+            except Exception as e:
+                problem_handler("exit", "Failed to apply artifact data to LIMS. Possibly due to data in laneBarcode.html; {}".format(e.message))
             
-        #If undetermined reads are greater than threshold*reads_in_lane
-        found_undet = round(float(undet_lane_reads)/(assign_lane_reads+undet_lane_reads)*100, 2)
-        if  found_undet > demux_process.udf["Maximum % Undetermined Reads per Lane"]:
-            #This looks kind of bad in lims, but \n can't be used.
-            problem_handler("warning", "Undemultiplexed reads for lane {} was {} ({})% thus exceeding \
-             defined limit.\t".format(lane_no, undet_lane_reads, found_undet))
-        else:
-            logger.info("Found {} ({}%) undemultiplexed reads for lane {}.".format(undet_lane_reads, found_undet, lane_no))
+        if not undet_included:
+            #If undetermined reads are greater than threshold*reads_in_lane
+            found_undet = round(float(undet_lane_reads)/(assign_lane_reads+undet_lane_reads)*100, 2)
+            if  found_undet > demux_process.udf["Maximum % Undetermined Reads per Lane"]:
+                #This looks kind of bad in lims, but \n can't be used.
```

Comment:
If you want to write long and comprehensive errrors, you can also wrtite to a log file and upload it.